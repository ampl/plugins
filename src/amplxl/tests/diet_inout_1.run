option solver minos;

option display_precision 0;
model diet.mod;

param dname symbolic := "amplxl";
param fname_in symbolic := "diet_in_1.xlsx";
param fname_out symbolic := "diet_inout_1.xlsx";

table Amounts IN (dname) (fname_in):
    [NUTR,FOOD], amt;
table Foods IN (dname) (fname_in):
    FOOD <- [FOOD], cost, f_min, f_max;
table Nutrients IN (dname) (fname_in):
    NUTR <- [NUTR], n_min, n_max;

read table Foods;
read table Nutrients;
read table Amounts;

display NUTR;
display FOOD;
display amt;

solve;

display {j in FOOD} (Buy[j], Buy.rc[j],  Buy[j]/f_max[j]);

table ExportFoods (dname) (fname_out) "Foods":
    [FOOD] IN, Buy OUT, Buy.rc ~ BuyRC OUT, {j in FOOD} Buy[j]/f_max[j] ~ BuyFrac OUT;
write table ExportFoods;

# read exported table to validate results
reset;
option display_precision 0;

param dname symbolic := "amplxl";
param fname_out symbolic := "diet_inout_1.xlsx";

set FOOD;
param cost{FOOD};
param f_min{FOOD};
param f_max{FOOD};
param Buy{FOOD};
param BuyFrac{FOOD};
param BuyRC{FOOD};

table Foods IN (dname) (fname_out):
    FOOD <- [FOOD], cost, f_min, f_max, Buy, BuyFrac, BuyRC;

read table Foods;

display FOOD;
display cost;
display f_min;
display f_max;
display Buy;
display BuyFrac;
display BuyRC;
