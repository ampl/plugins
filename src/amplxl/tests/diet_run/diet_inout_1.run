
model diet.mod;

param fname_in symbolic := "diet_xlsx/diet_in_1.xlsx";
param fname_out symbolic := "diet_xlsx/diet_inout_1.xlsx";

table Amounts IN "amplxl" (fname_in): 
    [NUTR,FOOD], amt;
table Foods IN "amplxl" (fname_in):
    FOOD <- [FOOD], cost, f_min, f_max;
table Nutrients IN "amplxl" (fname_in):
    NUTR <- [NUTR], n_min, n_max;

read table Foods;
read table Nutrients;
read table Amounts;

display NUTR;
display FOOD;
display amt;

solve;

table ExportFoods "amplxl" (fname_out) "Foods":
    [FOOD], Buy INOUT;
write table ExportFoods;
