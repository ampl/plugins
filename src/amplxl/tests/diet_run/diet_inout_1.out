set NUTR := A C B1 B2 NA CAL;

set FOOD := BEEF CHK FISH HAM MCH MTL SPG TUR;

amt [*,*] (tr)
:      A    B1   B2   C    CAL    NA     :=
BEEF   60   10   15   20   295    938
CHK     8   20   20    0   770   2180
FISH    8   15   10   10   440    945
HAM    40   35   10   40   430    278
MCH    15   15   15   35   315   1182
MTL    70   15   15   30   400    896
SPG    25   25   15   50   370   1329
TUR    60   15   10   20   450   1397
;

MINOS 5.51: optimal solution found.
13 iterations, objective 118.0594032
set FOOD := TUR SPG MTL MCH HAM FISH CHK BEEF;

cost [*] :=
BEEF  2.49
 CHK  1.99
FISH  1.99
 HAM  1.89
 MCH  2.89
 MTL  2.29
 SPG  2.59
 TUR  3.19
;

f_min [*] :=
BEEF  2
 CHK  2
FISH  2
 HAM  2
 MCH  2
 MTL  2
 SPG  2
 TUR  2
;

f_max [*] :=
BEEF  10
 CHK  10
FISH  10
 HAM  10
 MCH  10
 MTL  10
 SPG  10
 TUR  10
;

Buy [*] :=
BEEF   5.36061
 CHK   2
FISH   2
 HAM  10
 MCH  10
 MTL  10
 SPG   9.30605
 TUR   2
;

BuyFrac [*] :=
BEEF  0.536061
 CHK  0.2
FISH  0.2
 HAM  1
 MCH  1
 MTL  1
 SPG  0.930605
 TUR  0.2
;

BuyRC [*] :=
BEEF   2.22045e-15
 CHK   1.18884
FISH   1.14441
 HAM  -0.302651
 MCH  -0.551151
 MTL  -1.3289
 SPG   0
 TUR   2.73162
;

